### PR TITLE
Add assertion & avoid useless controller creation

### PR DESCRIPTION
**Pull request by ocrickard**
Fixes https://github.com/facebook/componentkit/issues/640 and is more
defensive with the resolution assertion.

### CODE REVIEW

File: ComponentKit/Core/Scope/CKComponentScopeHandle.mm [hunk 1]
```diff
@@ -152,7 +152,10 @@ - (BOOL)acquireFromComponent:(CKComponent *)component
 
 - (void)resolve
 {
-  if (!_controller) {
+  CKAssertFalse(_resolved);
```
Comment (eczarny) [inline ComponentKit/Core/Scope/CKComponentScopeHandle.mm]: A bit of a bummer that this assertion doesn't take a description.

**Comment by ocrickard** [comment]
So! Turns out this also fixes the problem internally with early-returning causing no controller creation later. I'm adding a unit test to confirm before landing.

**Comment by ocrickard** [comment]
Confirmed, the added test fails without this patch applied, and passes when it succeeds.

**Comment by eczarny** [comment]
Awesome! Feel free to merge once CI gives the go ahead.

**Pull request merged**